api/app/Http/Controllers/EntityController.php

```diff
@@ -91,15 +106,15 @@ public function postOne(Request $request)
      */
     public function putOne($id, Request $request)
     {
-        $this->validateId($id, $this->getKeyName(), $this->validateRequestRule);
+        $this->validateId($id, $this->getModel()->getKeyName(), $this->validateRequestRule);
 
         try {
-            $model = $this->getRepository()->find($id);
+            $model = $this->getEntityById($id);
         } catch (ModelNotFoundException $e) {
-            $model = $this->getRepository()->getNewModel();
+            $model = $this->getModel()->newInstance();
         }
         $model->fill($request->all());
-        $this->getRepository()->save($model);
+        $model->push();
```

Comment:
For this base controller, yes. Extending controllers might use push, but they will do that explicitly.